plugins/resin/src/main/java/com/navercorp/pinpoint/plugin/resin/interceptor/WebAppInterceptor.java

```diff
@@ -61,127 +50,75 @@ public void after(Object target, Object[] args, Object result, Throwable throwab
 			try {
 				String contextKey = extractContextKey(webapp);
 				List<String> loadedJarNames = extractLibJars(webapp);
-				if(isDebug){
-				logger.debug("{}  jars : {}", contextKey, Arrays.toString(loadedJarNames.toArray()));
+				if (isDebug) {
+					logger.debug("{}  jars : {}", contextKey, Arrays.toString(loadedJarNames.toArray()));
 				}
-				dispatchLibJars(contextKey, loadedJarNames,webapp);
+				dispatchLibJars(contextKey, loadedJarNames, webapp);
 			} catch (Exception e) {
-					logger.warn(e.getMessage(), e);
+				logger.warn(e.getMessage(), e);
 			}
 		} else {
 			logger.warn("Webapp loader is not an instance of org.apache.catalina.loader.WebappLoader. Found [{}]", target.getClass().toString());
 		}
 	}
 
 	private String extractContextKey(WebApp webapp) {
-		return webapp.getContextPath();
+		String context =  webapp.getContextPath();
+		return (context == null || context.length() == 0) ? "/ROOT": context;
 	}
-	
+
 	private List<String> extractLibJars(WebApp webapp) {
 		EnvironmentClassLoader classLoader = webapp.getEnvironmentClassLoader();
```

Comment:
If you change `WebApp` to `ServletContext`, this part will break as `ServletContext` does not have `getEnvironmentClassLoader()` method.

You can fix this by doing something like:
```
private List<String> extractLibJars(ServletContext webapp) {
    ClassLoader classLoader = webapp.getClassLoader();
    if (classLoader instanceof URLClassLoader) {
        URL[] urls = ((URLClassLoader) classLoader).getURLs();
        return extractLibJarNamesFromURLs(urls);
    } else {
        return Collections.emptyList();
    }
}
```